third_party/gpus/rocm_configure.bzl

```diff
@@ -255,6 +255,21 @@ def _hipcc_env(repository_ctx):
                          repository_ctx.os.environ[name].strip() + "\";")
     return hipcc_env.strip()
 
+def _hipcc_is_hipclang(repository_ctx):
+    """Returns if hipcc is based on hip-clang toolchain.
+
+    Args:
+        repository_ctx: The repository context.
+
+    Returns:
+        A string "True" if hipcc is based on hip-clang toolchain.
+        The functions returns "False" if not (ie: based on HIP/HCC toolchain).
+    """
+    for name in ["HIP_CLANG_PATH", "HIP_VDI_HOME"]:
```

Comment:
@whchung 
What happens, in general, is:  while building HIP RUNTIME, by default, it assumes that user will be going to use hcc compiler toolchain, for building the hip application, and accordingly, it creates below entry in the file - /opt/rocm/hip/lib/.hipInfo
-----
HIP_COMPILER=hcc
-----

However, while building HIP RUNTiME, if we explicitly pass the cmake option -DHIP_COMPILER=clang, then HIP RUNTIME assumes that the user will be going to use hip-clang compiler to build hip applications, and hence it makes below entry in the file -  /opt/rocm/hip/lib/.hipInfo 
----
HIP_COMPILER=clang
----

And, HIP does not automatically set neither HIP_CLANG_PATH nor HIP_VDI_HOME. These are for users to explicitly tell the system which compiler toolchain and runtime is being used.

Hence, typically,  as @yxsamliu suggested, we can grep for the string "HIP_COMPILER=clang" in the file /opt/rocm/hip/lib/.hipInfo and return True based on it, even before entering the "for loop" to check for the environment variables ["HIP_CLANG_PATH", "HIP_VDI_HOME"]

@yxsamliu Please confirm above.

Reply:
@mshivama thanks. I'm wondering would it be possible for `hipcc` to set the env vars automatically, based on `HIP_COMPILER` flag in `/opt/rocm/hip/lib/.hipInfo`.

In the updated PR you can see I have to spawn external process to "grep". This is not a good practice in general and certainly not portable to other platforms.

Reply:
> @mshivama thanks. I'm wondering would it be possible for `hipcc` to set the env vars automatically, based on `HIP_COMPILER` flag in `/opt/rocm/hip/lib/.hipInfo`.
> 
> In the updated PR you can see I have to spawn external process to "grep". This is not a good practice in general and certainly not portable to other platforms.

@whchung crosstool launches hipcc to compile HIP program, not hipcc crosstool. Therefore even if hipcc sets the environment variable, crosstool cannot see it. I think the source of the this issue is that we have two compilers and we need to differentiate them. Once HCC is deprecated, we only need to support hip-clang, then we do not need to check this.